opm/core/simulator/WellState.hpp

```diff
@@ -124,7 +124,7 @@ namespace Opm
                                 wellrates_[np*w + p] = rate_target * distr[p];
                             }
                         } else {
-                            const double small_rate = 1e-14;
```

Comment:
I do not understand why this matters? could you please give some explanations?

Reply:
If the well rate is zero the mix[..] will be set to comp_frac[..] instead of 1/3. 

#line 1398ff in StandardWell_impl.hpp
---
            if (tot_surf_rate != 0.0) {
                for (int component = 0; component < num_comp; ++component) {
                    mix[component] = std::fabs(q_out_perf[perf*num_comp + component]/tot_surf_rate);
                }
            } else {
                // No flow => use well specified fractions for mix.
                for (int component = 0; component < num_comp; ++component) {
                    if (component < np) {
                        mix[component] = comp_frac_[ ebosCompIdxToFlowCompIdx(component)];
                    }
                }
                // intialize 0.0 for comIdx >= np;
            }
---

Reply:
This is good.

Reply:
Could this mess up some legacy code?